caste.go

```diff
@@ -260,7 +263,7 @@ func ToInt64E(i interface{}) (int64, error) {
 	case float32:
 		return int64(s), nil
 	case string:
-		v, err := strconv.ParseInt(trimZeroDecimal(s), 0, 0)
+		v, err := strconv.ParseInt(trimDecimal(s), 0, 0)
```

Comment:
Technically, it may be more correct to parse into float when a dot is present and convert to int after that.